interfaces/apparmor/spec_test.go

```diff
@@ -144,26 +144,141 @@ apps:
   vanguard:
     command: vanguard
 layout:
-  /usr:
-    bind: $SNAP/usr
-  /mytmp:
+  /usr/foo:
+    bind: $SNAP/usr/foo
+  /var/tmp:
     type: tmpfs
     mode: 1777
-  /mylink:
-    symlink: $SNAP/link/target
+  /var/cache/mylink:
+    symlink: $SNAP_DATA/link/target
   /etc/foo.conf:
     bind-file: $SNAP/foo.conf
 `
 
 func (s *specSuite) TestApparmorSnippetsFromLayout(c *C) {
 	snapInfo := snaptest.MockInfo(c, snapWithLayout, &snap.SideInfo{Revision: snap.R(42)})
+	restore := apparmor.SetSpecScope(s.spec, []string{"snap.vanguard.vanguard"}, "vanguard")
+	defer restore()
+
 	s.spec.AddSnapLayout(snapInfo)
 	c.Assert(s.spec.Snippets(), DeepEquals, map[string][]string{
 		"snap.vanguard.vanguard": {
 			"# Layout path: /etc/foo.conf\n/etc/foo.conf mrwklix,",
-			"# Layout path: /mylink\n# (no extra permissions required for symlink)",
-			"# Layout path: /mytmp\n/mytmp{,/**} mrwklix,",
-			"# Layout path: /usr\n/usr{,/**} mrwklix,",
+			"# Layout path: /usr/foo\n/usr/foo{,/**} mrwklix,",
+			"# Layout path: /var/cache/mylink\n# (no extra permissions required for symlink)",
+			"# Layout path: /var/tmp\n/var/tmp{,/**} mrwklix,",
 		},
 	})
+
+	profile0 := `  # Layout /etc/foo.conf: bind-file $SNAP/foo.conf
+  mount options=(bind, rw) /snap/vanguard/42/foo.conf -> /etc/foo.conf,
+  umount /etc/foo.conf,
+  # Writable mimic /etc
```

Comment:
Commenting here rather than above as it's easier to see the context. I think adjusting this comment to be:
```# Writable mimic /etc (/tmp/.snap and /etc are in per-snap mount namespace)```

Otherwise someone might think there are symlinks attacks on the global (ie, host's) /tmp or wonder why it is allowed to umount /etc. I don't think these are needed for the SNAP* dirs, below since the rules are more clear there. This could be handle via a code comment instead of a policy comment if that is easier.

Reply:
This isn't needed if you add the requested comment, above.